### PR TITLE
Clang-inspired cleanups; abs vs fabsf, force correct use of override keyword

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by peterbarker** [comment]
I intend to rebase this after some GCS work has been negating the need for adding override to the sensor-status-flags calls.

File: Tools/ardupilotwaf/boards.py [hunk 1]
```diff
@@ -176,7 +176,7 @@ def configure_env(self, cfg, env):
                 '-fcolor-diagnostics',
 
                 '-Wno-gnu-designator',
-                '-Wno-inconsistent-missing-override',
+                '-Werror=inconsistent-missing-override',
```
Comment (OXINARF) [inline Tools/ardupilotwaf/boards.py]: Can we separate the forced error from the disabled warnings?

Comment (peterbarker) [reply Tools/ardupilotwaf/boards.py]: As in re-order things?

**Comment by lvale** [comment]
should these be fixed too ??

>[ 24/532] Compiling libraries/AC_PrecLand/AC_PrecLand_SITL_Gazebo.cpp
../../libraries/AC_AutoTune/AC_AutoTune.cpp:441:22: warning: using floating point absolute value function 'fabsf' when argument is of integer type [-Wabsolute-value]
                     fabsf(ahrs_view->roll_sensor - roll_cd),
                     ^
../../libraries/AC_AutoTune/AC_AutoTune.cpp:441:22: note: use function 'std::abs' instead
                     fabsf(ahrs_view->roll_sensor - roll_cd),
                     ^~~~~
                     std::abs
../../libraries/AC_AutoTune/AC_AutoTune.cpp:447:22: warning: using floating point absolute value function 'fabsf' when argument is of integer type [-Wabsolute-value]
                     fabsf(ahrs_view->pitch_sensor - pitch_cd),
                     ^
../../libraries/AC_AutoTune/AC_AutoTune.cpp:447:22: note: use function 'std::abs' instead
                     fabsf(ahrs_view->pitch_sensor - pitch_cd),
                     ^~~~~
                     std::abs
2 warnings generated.


In file included from ../../libraries/AP_Compass/AP_Compass_RM3100.cpp:20:
../../libraries/AP_Compass/AP_Compass_RM3100.h:55:14: warning: private field 'accum_count' is not used [-Wunused-private-field]
    uint16_t accum_count;
             ^
1 warning generated.


[139/532] Compiling libraries/AP_InertialSensor/AP_InertialSensor_SITL.cpp
../../libraries/AP_IRLock/AP_IRLock_I2C.cpp:41:11: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
    dev = std::move(hal.i2c_mgr->get_device(bus, IRLOCK_I2C_ADDRESS));
          ^
../../libraries/AP_IRLock/AP_IRLock_I2C.cpp:41:11: note: remove std::move call here
    dev = std::move(hal.i2c_mgr->get_device(bus, IRLOCK_I2C_ADDRESS));
          ^~~~~~~~~~                                                ~
1 warning generated.


[169/532] Compiling libraries/AP_Math/vector2.cpp
../../libraries/AP_Math/AP_Math.cpp:33:12: warning: absolute value function 'fabsf' given an argument of type 'double' but has parameter of type 'float' which may cause truncation of value [-Wabsolute-value]
    return fabsf(v_1 - v_2) < std::numeric_limits<float>::epsilon();
           ^
../../libraries/AP_Math/AP_Math.cpp:40:15: note: in instantiation of function template specialization 'is_equal<double, double>' requested here
template bool is_equal<double>(const double v_1, const double v_2);
              ^
../../libraries/AP_Math/AP_Math.cpp:33:12: note: use function 'std::abs' instead
    return fabsf(v_1 - v_2) < std::numeric_limits<float>::epsilon();
           ^~~~~
           std::abs
1 warning generated.


[232/532] Compiling libraries/AP_Notify/RCOutputRGBLed.cpp
../../libraries/AP_Notify/OreoLED_I2C.cpp:60:12: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
    _dev = std::move(hal.i2c_mgr->get_device(_bus, OREOLED_BASE_I2C_ADDR));
           ^
../../libraries/AP_Notify/OreoLED_I2C.cpp:60:12: note: remove std::move call here
    _dev = std::move(hal.i2c_mgr->get_device(_bus, OREOLED_BASE_I2C_ADDR));
           ^~~~~~~~~~                                                    ~
1 warning generated.

[233/532] Compiling libraries/AP_Notify/DiscoLED.cpp
[234/532] Compiling libraries/AP_Notify/Display_SH1106_I2C.cpp
[235/532] Compiling libraries/AP_Notify/PCA9685LED_I2C.cpp
[236/532] Compiling libraries/AP_Notify/Display.cpp
[237/532] Compiling libraries/AP_Notify/Buzzer.cpp
[238/532] Compiling libraries/AP_Notify/DiscreteRGBLed.cpp
../../libraries/AP_Notify/NCP5623.cpp:69:16: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
        _dev = std::move(hal.i2c_mgr->get_device(_bus, addrs[i]));
               ^
../../libraries/AP_Notify/NCP5623.cpp:69:16: note: remove std::move call here
        _dev = std::move(hal.i2c_mgr->get_device(_bus, addrs[i]));
               ^~~~~~~~~~                                       ~
1 warning generated.

[239/532] Compiling libraries/AP_OSD/AP_OSD_SITL.cpp
[240/532] Compiling libraries/AP_OSD/AP_OSD_Screen.cpp
[241/532] Compiling libraries/AP_OSD/AP_OSD_Setting.cpp
../../libraries/AP_Notify/ToshibaLED_I2C.cpp:51:12: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
    _dev = std::move(hal.i2c_mgr->get_device(_bus, TOSHIBA_LED_I2C_ADDR));
           ^
../../libraries/AP_Notify/ToshibaLED_I2C.cpp:51:12: note: remove std::move call here
    _dev = std::move(hal.i2c_mgr->get_device(_bus, TOSHIBA_LED_I2C_ADDR));
           ^~~~~~~~~~                                                   ~
1 warning generated.

[242/532] Compiling libraries/AP_OSD/AP_OSD_Backend.cpp
[243/532] Compiling libraries/AP_OSD/AP_OSD.cpp
[244/532] Compiling libraries/AP_OSD/AP_OSD_MAX7456.cpp
[245/532] Compiling libraries/AP_OpticalFlow/OpticalFlow_backend.cpp
[246/532] Compiling libraries/AP_OpticalFlow/AP_OpticalFlow_Onboard.cpp
[247/532] Compiling libraries/AP_OpticalFlow/AP_OpticalFlow_Pixart.cpp
../../libraries/AP_Notify/Display.cpp:329:50: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
            _driver = Display_SSD1306_I2C::probe(std::move(hal.i2c_mgr->get_device(i, NOTIFY_DISPLAY_I2C_ADDR)));
                                                 ^
../../libraries/AP_Notify/Display.cpp:329:50: note: remove std::move call here
            _driver = Display_SSD1306_I2C::probe(std::move(hal.i2c_mgr->get_device(i, NOTIFY_DISPLAY_I2C_ADDR)));
                                                 ^~~~~~~~~~                                                   ~
../../libraries/AP_Notify/Display.cpp:333:49: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
            _driver = Display_SH1106_I2C::probe(std::move(hal.i2c_mgr->get_device(i, NOTIFY_DISPLAY_I2C_ADDR)));
                                                ^
../../libraries/AP_Notify/Display.cpp:333:49: note: remove std::move call here
            _driver = Display_SH1106_I2C::probe(std::move(hal.i2c_mgr->get_device(i, NOTIFY_DISPLAY_I2C_ADDR)));
                                                ^~~~~~~~~~                                                   ~
2 warnings generated.

[248/532] Compiling libraries/AP_OpticalFlow/AP_OpticalFlow_CXOF.cpp
[249/532] Compiling libraries/AP_OpticalFlow/AP_OpticalFlow_PX4Flow.cpp
[250/532] Compiling libraries/AP_OpticalFlow/AP_OpticalFlow_SITL.cpp
[251/532] Compiling libraries/AP_OpticalFlow/OpticalFlow.cpp
../../libraries/AP_OSD/AP_OSD.cpp:167:50: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
        AP_HAL::OwnPtr<AP_HAL::Device> spi_dev = std::move(hal.spi->get_device("osd"));
                                                 ^
../../libraries/AP_OSD/AP_OSD.cpp:167:50: note: remove std::move call here
        AP_HAL::OwnPtr<AP_HAL::Device> spi_dev = std::move(hal.spi->get_device("osd"));
                                                 ^~~~~~~~~~                          ~
1 warning generated.

[252/532] Compiling libraries/AP_Parachute/AP_Parachute.cpp
[253/532] Compiling libraries/AP_Param/AP_Param.cpp
[254/532] Compiling libraries/AP_Proximity/AP_Proximity_SITL.cpp
[255/532] Compiling libraries/AP_Proximity/AP_Proximity_RPLidarA2.cpp
../../libraries/AP_OpticalFlow/AP_OpticalFlow_Pixart.cpp:86:12: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
    _dev = std::move(hal.spi->get_device(devname));
           ^
../../libraries/AP_OpticalFlow/AP_OpticalFlow_Pixart.cpp:86:12: note: remove std::move call here
    _dev = std::move(hal.spi->get_device(devname));
           ^~~~~~~~~~                            ~
1 warning generated.


In file included from ../../libraries/AP_Proximity/AP_Proximity_TeraRangerTowerEvo.cpp:17:
../../libraries/AP_Proximity/AP_Proximity_TeraRangerTowerEvo.h:56:19: warning: private field 'SEQUENCE_MODE' is not used [-Wunused-private-field]
    const uint8_t SEQUENCE_MODE[4] = {(uint8_t)0x00, (uint8_t)0x31, (uint8_t)0x02, (uint8_t)0xE2};
                  ^
../../libraries/AP_Proximity/AP_Proximity_TeraRangerTowerEvo.h:58:19: warning: private field 'REFRESH_50_HZ' is not used [-Wunused-private-field]
    const uint8_t REFRESH_50_HZ[5] = { (uint8_t)0x00, (uint8_t)0x52, (uint8_t)0x03, (uint8_t)0x02, (uint8_t)0xC3};
                  ^
../../libraries/AP_Proximity/AP_Proximity_TeraRangerTowerEvo.h:60:19: warning: private field 'REFRESH_250_HZ' is not used [-Wunused-private-field]
    const uint8_t REFRESH_250_HZ[5] = { (uint8_t)0x00, (uint8_t)0x52, (uint8_t)0x03, (uint8_t)0x04, (uint8_t)0xD1};
                  ^
../../libraries/AP_Proximity/AP_Proximity_TeraRangerTowerEvo.h:61:19: warning: private field 'REFRESH_500_HZ' is not used [-Wunused-private-field]
    const uint8_t REFRESH_500_HZ[5] = { (uint8_t)0x00, (uint8_t)0x52, (uint8_t)0x03, (uint8_t)0x05, (uint8_t)0xD6};
                  ^
../../libraries/AP_Proximity/AP_Proximity_TeraRangerTowerEvo.h:62:19: warning: private field 'REFRESH_600_HZ' is not used [-Wunused-private-field]
    const uint8_t REFRESH_600_HZ[5] = { (uint8_t)0x00, (uint8_t)0x52, (uint8_t)0x03, (uint8_t)0x06, (uint8_t)0xDF};
                  ^
5 warnings generated.


[353/532] Compiling libraries/SRV_Channel/SRV_Channel_aux.cpp
In file included from ../../libraries/GCS_MAVLink/GCS_MAVLink.cpp:36:
libraries/GCS_MAVLink/include/mavlink/v2.0/mavlink_helpers.h:247:25: warning: taking address of packed member 'checksum' of class or structure '__mavlink_message' may result in an unaligned pointer value [-Waddress-of-packed-member]
        crc_accumulate_buffer(&msg->checksum, _MAV_PAYLOAD(msg), msg->len);
                               ^~~~~~~~~~~~~
libraries/GCS_MAVLink/include/mavlink/v2.0/mavlink_helpers.h:248:29: warning: taking address of packed member 'checksum' of class or structure '__mavlink_message' may result in an unaligned pointer value [-Waddress-of-packed-member]
        crc_accumulate(crc_extra, &msg->checksum);
                                   ^~~~~~~~~~~~~
libraries/GCS_MAVLink/include/mavlink/v2.0/mavlink_helpers.h:471:12: warning: taking address of packed member 'checksum' of class or structure '__mavlink_message' may result in an unaligned pointer value [-Waddress-of-packed-member]
        crc_init(&msg->checksum);
                  ^~~~~~~~~~~~~
libraries/GCS_MAVLink/include/mavlink/v2.0/mavlink_helpers.h:476:21: warning: taking address of packed member 'checksum' of class or structure '__mavlink_message' may result in an unaligned pointer value [-Waddress-of-packed-member]
        crc_accumulate(c, &msg->checksum);
                           ^~~~~~~~~~~~~
4 warnings generated.


[409/532] Compiling libraries/SITL/SITL.cpp
../../libraries/AP_HAL_SITL/sitl_gps.cpp:485:17: warning: cast from 'int32_t *' (aka 'int *') to 'uint32_t *' (aka 'unsigned int *') increases required alignment from 1 to 4 [-Wcast-align]
    swap_uint32((uint32_t *)&p.latitude, 5);
                ^~~~~~~~~~~~~~~~~~~~~~~
../../libraries/AP_HAL_SITL/sitl_gps.cpp:485:30: warning: taking address of packed member 'latitude' of class or structure 'mtk_msg' may result in an unaligned pointer value [-Waddress-of-packed-member]
    swap_uint32((uint32_t *)&p.latitude, 5);
                             ^~~~~~~~~~
../../libraries/AP_HAL_SITL/sitl_gps.cpp:486:30: warning: taking address of packed member 'utc_time' of class or structure 'mtk_msg' may result in an unaligned pointer value [-Waddress-of-packed-member]
    swap_uint32((uint32_t *)&p.utc_time, 1);
                             ^~~~~~~~~~
3 warnings generated.

[410/532] Compiling libraries/SITL/SIM_Sailboat.cpp
../../libraries/SITL/SIM_XPlane.cpp:141:29: warning: cast from 'uint8_t *' (aka 'unsigned char *') to 'const float *' increases required alignment from 1 to 4 [-Wcast-align]
        const float *data = (const float *)p;
                            ^~~~~~~~~~~~~~~~
1 warning generated.


[447/532] Compiling libraries/AP_Rally/AP_Rally.cpp
../../libraries/AP_Baro/AP_Baro.cpp:602:43: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                          std::move(hal.i2c_mgr->get_device(_ext_bus, HAL_BARO_MS5611_I2C_ADDR))));
                                          ^
../../libraries/AP_Baro/AP_Baro.cpp:639:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(hal.i2c_mgr->get_device(i, HAL_BARO_BMP085_I2C_ADDR))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:645:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(hal.i2c_mgr->get_device(i, HAL_BARO_BMP280_I2C_ADDR))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:647:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(hal.i2c_mgr->get_device(i, HAL_BARO_BMP280_I2C_ADDR2))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:653:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(hal.i2c_mgr->get_device(i, HAL_BARO_MS5611_I2C_ADDR))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:655:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(hal.i2c_mgr->get_device(i, HAL_BARO_MS5611_I2C_ADDR2))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:661:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(hal.i2c_mgr->get_device(i, HAL_BARO_MS5607_I2C_ADDR)),
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:668:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(hal.i2c_mgr->get_device(i, HAL_BARO_MS5637_I2C_ADDR)),
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:675:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(hal.i2c_mgr->get_device(i, HAL_BARO_FBM320_I2C_ADDR))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:677:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(hal.i2c_mgr->get_device(i, HAL_BARO_FBM320_I2C_ADDR2))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:683:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(hal.i2c_mgr->get_device(i, HAL_BARO_DPS280_I2C_ADDR))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:685:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(hal.i2c_mgr->get_device(i, HAL_BARO_DPS280_I2C_ADDR2))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:691:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(hal.i2c_mgr->get_device(i, HAL_BARO_LPS25H_I2C_ADDR))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:697:49: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                                std::move(hal.i2c_mgr->get_device(i, HAL_BARO_KELLERLD_I2C_ADDR))));
                                                ^
../../libraries/AP_Baro/AP_Baro.cpp:703:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(hal.i2c_mgr->get_device(i, HAL_BARO_MS5837_I2C_ADDR)), AP_Baro_MS56XX::BARO_MS5837));
                                              ^
15 warnings generated.

**Comment by peterbarker** [comment]
@lvale the autotune errors should be fixed by this PR.

The moving-reference thingy - not so much :-)  That would take some time to get one's head around.

I'm tempted to make the abs thing fatal - they seem to get endlessly ignored otherwise...

File: Tools/ardupilotwaf/boards.py (hunk 1 above)
Comment (peterbarker) [reply Tools/ardupilotwaf/boards.py]: Assumed this was reordering things....

**Pull request merged**